apps/functions-subscription/src/use-cases/create-trial.ts

```diff
@@ -0,0 +1,16 @@
+import * as RTE from 'fp-ts/ReaderTaskEither';
+import { makeTrial, Trial } from '../domain/trial';
+import { pipe } from 'fp-ts/lib/function';
+import { Capabilities } from '../domain/capabilities';
+
+export const createTrial = (
+  name: Trial['name'],
+  description: Trial['description'],
+) =>
+  pipe(
+    RTE.ask<Pick<Capabilities, 'trialWriter'>>(),
+    RTE.apSW('trial', makeTrial(name, description)),
+    RTE.flatMapTaskEither(({ trial, trialWriter }) =>
+      trialWriter.insert(trial),
+    ),
+  );
```

Comment:
**suggestion (non-blocking)**: The use case seems really trivial. What about adding an `insertTrial` method into `domain/trial.ts` file that, given a name and a description, returns a trial? Doing so, we can remove the use-case and use the insert directly.

Reply:
I'd keep the use case at the moment.